### ISSUE TITLE
Animate Shader's uniform value

### ISSUE THREAD

**Issue by EhsanFortuna** [godotengine/godot#73770]
### Godot version

4.0 RC

### System information

Linux 

### Issue description

when adding new keyframes from the uniform's variable, value is always set to 0 . 

### Steps to reproduce

1 - create a new sprite and add shader material
2 - in shader add uniform 
3 - add multiple keys from uniform to animation player
4 - keyframe's value does not change

### Minimal reproduction project

[demo.zip](https://github.com/godotengine/godot/files/10806409/demo.zip)

**Comment by clayjohn** [comment]
When you add new keyframes the keyframe is set to whatever the variable is set to. In your MRP the uniform is set to ``0.0``, so when you add a new keyframe it adds it with value 0.0. 

If you change the uniform to 0.5 and add a new keyframe it adds it with value 0.5. 

I have tested your MRP locally and adding keyframes is working fine.

**Comment by EhsanFortuna** [comment]
when changing the values of uniform and clicking on "add new keyframe" the new value does not save, 
you should press enter to save it, 

I get it now, but it would be nice I guess to automate the saving, after all, I'm going to add a new keyframe

**Comment by clayjohn** [comment]
> when changing the values of uniform and clicking on "add new keyframe" the new value does not save, you should press enter to save it,
> 
> I get it now, but it would be nice I guess to automate the saving, after all, I'm going to add a new keyframe

I'm sorry, I don't understand what you mean. Are you saying that you have to press enter after entering the number in the uniform before clicking on the add new keyframe button?

**Comment by EhsanFortuna** [comment]
> > when changing the values of uniform and clicking on "add new keyframe" the new value does not save, you should press enter to save it,
> > I get it now, but it would be nice I guess to automate the saving, after all, I'm going to add a new keyframe
> 
> I'm sorry, I don't understand what you mean. Are you saying that you have to press enter after entering the number in the uniform before clicking on the add new keyframe button?

Yes, Actually I've never noticed it before, it happens on every property,  you have to save it by pressing enter and then add new key frame

**Comment by clayjohn** [comment]
Oh, in that case this has nothing to do with Animation or Keyframes. This is just how you input values into a text box, you need to click away or press enter for the value to be entered into the text box

**Comment by EhsanFortuna** [comment]
> Oh, in that case, this has nothing to do with Animation or Keyframes. This is just how you input values into a text box, you need to click away or press enter for the value to be entered into the text box
 
why wouldn't clicking "add new frame"  do something like "click away"?

**Comment by clayjohn** [comment]
> > Oh, in that case, this has nothing to do with Animation or Keyframes. This is just how you input values into a text box, you need to click away or press enter for the value to be entered into the text box
> 
> why wouldn't clicking "add new frame" do something like "click away"?

Its probably an order of operations issue, by clicking the add keyframe button the keyframe is probably getting added before the value is confirmed
